packages/sitecore-jss-react/src/components/PlaceholderMetadata.tsx

```diff
@@ -0,0 +1,85 @@
+﻿import React from 'react';
+import { ComponentRendering } from '@sitecore-jss/sitecore-jss/layout';
+
+/**
+ *  Props containing the component data to render.
+ */
+export interface PlaceholderMetadataProps {
+  component: ComponentRendering;
+}
+
+/**
+ * This component supports the chrome's hydration process in Pages by rendering the required metadata.
+ * This component handles recursive rendering of components and their placeholders, encapsulating
+ * each part within specific code tags to outline their structure and metadata attributes.
+ *
+ * @param {PlaceholderMetadataProps} props - The props containing the component data to render.
+ * @returns {JSX.Element} - The rendered component with all nested components and placeholders.
+ */
+export const PlaceholderMetadata = ({ component }: PlaceholderMetadataProps): JSX.Element => {
+  const codeAttributes = (chromeType: string, kind: string) => {
+    return {
+      type: 'text/sitecore',
+      chrometype: `${chromeType}`,
+      className: 'scpm',
+      kind: `${kind}`,
+    };
+  };
+
+  const renderComponent = (component: ComponentRendering): JSX.Element => {
+    const ComponentName = component.componentName as React.ElementType;
+    return (
+      <>
+        <code {...codeAttributes('rendering', 'open')}>{`{uid: "${component.uid}"}`}</code>
+        <ComponentName />
+        <code {...codeAttributes('rendering', 'close')}></code>
+      </>
+    );
+  };
+
+  const renderNestedComponents = (components: ComponentRendering[]): JSX.Element[] => {
```

Comment:
This function is kind of similar to _renderComponent_, I mean by name
should we encapsulate it into the _renderComponentsInPlaceholder_?

Reply:
I have done some refactoring, they don't look the same now. Its very hard to combine them into one function.